src/components/Modal/ModalHeader/ModalHeader.tsx

```diff
@@ -82,19 +87,20 @@ const ModalHeader: FC<ModalHeaderProps> = ({
 }) => {
   return (
     <div className={cx(styles.container, className)}>
-      <Heading
-        id={id}
-        maxLines={2}
-        data-testid={dataTestId || getTestId(ComponentDefaultTestId.MODAL_HEADER, id)}
-        className={titleClassName}
-      >
-        {icon && (
-          <span className={cx(styles.icon, iconClassName)}>
-            <Icon icon={icon} iconType={Icon.type.SVG} iconSize={iconSize} ignoreFocusStyle clickable={false} />
-          </span>
+      <div data-testid={dataTestId || getTestId(ComponentDefaultTestId.MODAL_HEADER, id)}>
```

Comment:
can't you simply set the data-testid on the root div of the component? do we really need this extra div?